### ISSUE TITLE
train_model.zip error

### ISSUE THREAD

**Issue by TuBieJun** [google/deepvariant#10]
When i run train_model.zip, i get a error:
![image](https://user-images.githubusercontent.com/15261087/33821907-8e07de38-de90-11e7-9461-dc54523691d4.png)
Here is  my command:
```
python /leostore/software/deepvariant/bazel-bin/deepvariant/make_examples.zip --dataset_config_pbtxt "/leostore/analysis/development/liteng/deepvariant_test/test_train.config.txt" --start_from_checkpoint inception_v3.ckpt
```

**Comment by pichuan** [comment]
Hi,
can you tell me where you got your inception_v3.ckpt* model files?
And, can you paste the content of your test_train.config.txt file?

**Comment by pichuan** [comment]
Oh!! By the way, if you want to train a model, you shouldn't be using make_examples.zip (like you listed in your example). You should be using: model_train.zip.

Here is an example of a latest small training run that I did:
python "${BIN_DIR}"/model_train.zip \
--dataset_config_pbtxt /home/pichuan/training/training.dataset_config.pbtxt \
--train_dir /home/pichuan/train_dir \
--start_from_checkpoint /home/pichuan/inception_v3_ckpt/model.ckpt

My training.dataset_config.pbtxt looks like this:
name: "small_dataset"
tfrecord_path: "/home/pichuan/training/small_dataset.training.shuffled.examples-?????-of-00010.tfrecord.gz"
num_examples: 72178

**Comment by TuBieJun** [comment]
Sorry,  i used model_train.zip really,  just pasted the incorrect command to issue. 
This is my correct command:
```
python /leostore/software/deepvariant/bazel-bin/deepvariant/model_train.zip --dataset_config_pbtxt "/leostore/analysis/development/liteng/deepvariant_test/test_train.config.txt" --start_from_checkpoint inception_v3.ckpt
```
The inception_v3.ckpt is downloaded from https://github.com/tensorflow/models/tree/master/research/slim#Data 
This is my  config file:
```
name: "test-training-dataset"
tfrecord_path: "/leostore/analysis/development/liteng/deepvariant_test/train_set/test_train.tfrecord.gz"
num_examples: 1
```

**Comment by pichuan** [comment]
I suggest that you try a dataset with more training examples.
Can you try providing a tfrecord_path with more examples (let's say, 100 or 1000 examples?) and provide the corresponding num_examples, and let us know if that works for you?

**Comment by TuBieJun** [comment]
I changed the "num_examples" form 1 to 2 only,not  add any  tfrecord.gz  file, then it start run, but it is really slow, it have run two hours stilly.  So is this situation normal? The tfrecord.gz file is about 2.4M size.

![image](https://user-images.githubusercontent.com/15261087/33869274-d250d54c-df42-11e7-9d37-a6cb401e4cdc.png)



Here my config.txt:
```
name: "test-training-dataset"
tfrecord_path: "/leostore/analysis/development/liteng/deepvariant_test/train_set/test_train.tfrecord.gz"
num_examples: 2
```
This is my command
```
python /leostore/software/deepvariant/bazel-bin/deepvariant/model_train.zip --dataset_config_pbtxt /leostore/analysis/development/liteng/deepvariant_test/test_train.config.txt --start_from_checkpoint inception_v3.ckpt
```

**Comment by pichuan** [comment]
First of all, the num_examples is meant to be how many tensorflow.Example are actually in the file pointed at tfrecord_path. You should set it correctly. (I assume you're not actually only training on 2 examples?)

Second, if you want to run training, I highly recommend that you use a machine with GPU.
See: https://github.com/google/deepvariant/blob/master/docs/deepvariant-details.md on how to build with GPU.

**Comment by TuBieJun** [comment]
The tfrecord file  i used  come from a sample data. By the way if i have many tfrecord files come from some different sample, what should i set "tfrecord_path"? Use wildcard like this "/leostore/analysis/development/liteng/deepvariant_test/train_set/*tfrecord.gz" or just use the "cat" command to combine all tfrecord files to a  tfrecord file ?

**Comment by pgrosu** [comment]
Based on the proto code at the link below, it seems to require full paths without wildcards:

https://github.com/google/deepvariant/blob/dba790c6f1618831a58b7230535ec14c323461d7/deepvariant/protos/deepvariant.proto#L528-L529

**Comment by pichuan** [comment]
If I remember correctly, wildcards like * and ? should work. We can probably improve the comment there.
But concatenating everything together works too. You can directly cat all `*tfrecord.gz` into another big all.tfrecord.gz file. I would suggest trying wildcard first though.

In terms of how to set num_examples: for now if you know roughly how many examples you have (for example, I can't remember if make_examples print out that information), you can just set a rough number. It's only being used here: 
https://github.com/google/deepvariant/blob/r0.4/deepvariant/model_train.py#L211
It does affect the learning rate decay, but it doesn't have to be exact.
I'll see if I can come back with a better example to count examples later.

**Comment by pichuan** [comment]
Update:
if you look at https://github.com/google/deepvariant/blob/r0.5/docs/visualizing_examples.ipynb
the code in read_tfrecords is an example of how you can read a tfrecord file, and count examples if you like.
